src/Marten/IDocumentSchemaResolver.cs

```diff
@@ -0,0 +1,57 @@
+#nullable enable
+namespace Marten;
+
+public interface IDocumentSchemaResolver
+{
+    /// <summary>
+    ///     The schema name used to store the documents.
+    /// </summary>
+    string DatabaseSchemaName { get; }
+
+    /// <summary>
+    ///     The database schema name for event related tables. By default this
+    ///     is the same schema as the document storage
+    /// </summary>
+    string EventsSchemaName { get; }
+
+    /// <summary>
+    ///     Find the database name of the table backing <typeparamref name="TDocument"/>. Supports documents and projections.
+    /// </summary>
+    /// <typeparam name="TDocument">The document or projection to look up.</typeparam>
+    /// <param name="qualified" default="true">
+    ///     When true (default) the qualified name is returned (schema and table name).
+    ///     Otherwise only the table name is returned.
+    /// </param>
+    /// <returns>The name of <typeparamref name="TDocument"/> in the database.</returns>
+    string For<TDocument>(bool qualified = true);
```

Comment:
Could we add an overload that accepts Type as a parameter? I could see a scenario where someone wants to loop through an array of types to run sql against multiple tables.

Reply:
I'll do that in a bit. It's almost inevitable that using Method<T> means you'll eventually want Method(Type)